xbmc/cores/AudioEngine/Sinks/AESinkAUDIOTRACK.cpp

```diff
@@ -451,15 +450,15 @@ bool CAESinkAUDIOTRACK::Initialize(AEAudioFormat &format, std::string &device)
       switch (m_format.m_streamInfo.m_type)
       {
         case CAEStreamInfo::STREAM_TYPE_TRUEHD:
-          m_min_buffer_size = MAX_RAW_AUDIO_BUFFER_HD;
-          m_format.m_frames = m_min_buffer_size;
+          m_min_buffer_size = std::max(m_min_buffer_size * 2, MAX_RAW_AUDIO_BUFFER_HD);
+          m_format.m_frames = MAX_RAW_AUDIO_BUFFER_HD;
```

Comment:
In RAW frames == bytes always, then this not should be `m_min_buffer_size` ?

ie:
`m_format.m_frames = m_min_buffer_size;`

(not runtime tested yet)

Reply:
One is the buffer, which might be larger. But periods, which is in m_format.m_frames are the "size" sent at once. Ususally for a 10 ms packages, we use a buffer of 80 ms. With the intention that it won't run dry too fast. For RAW this logic is somehow flawed, as in general it makes not much sense to only have "one buffer" that is filled always fully with one AddPacket call.

As said, the above is just a test.

To complete that:
Period = sampleRate / 20
Buffer = 4 times period.

as what we normally do, also in IEC case for the audio sink.